lib/tasks/application.rake

```diff
@@ -121,4 +121,43 @@ namespace :application do
     Rails.logger = ActiveSupport::Logger.new($stdout)
     Rails.logger.level = 1
   end
+
+  namespace :links_valid do
+    desc "Checks the validity of links in division summary"
+    task divisions: :environment do
+      base_url = "https://theyvoteforyou.org.au"
+      md = Redcarpet::Markdown.new(Redcarpet::Render::HTML)
+
+      Division.find_each do |division|
+        wiki_motion = division.wiki_motion
+
+        if wiki_motion
+          summary_in_html = md.render(wiki_motion.description)
+          parsed_data = Nokogiri::HTML.parse(summary_in_html)
+          url_hash_table = Hash.new(0)
+
+          tags = parsed_data.xpath("//a")
+          tags.each do |tag|
+            url_from_page = tag[:href]
+            url = url_from_page
+
+            url = "http://#{url}" if !url[%r{\Ahttp://}] && !url[%r{\Ahttps://}] # adds http to the start of the URL to avoid failure
+
+            begin
+              if (url_hash_table[url]).zero?
+                url_hash_table[url] += 1
+                uri = URI(url)
+                Net::HTTP.get(uri)
+              end
+            rescue StandardError
```

Comment:
If a particular URL that is broken appears on several division pages won't this logic just show the first one that it errors on rather than showing all the division pages that the broken url is on (while at the same time only doing one web request)?